### PR TITLE
feat(mux-player): Seek-to-live button to use Media Chrome

### PR DESCRIPTION
**Pull request by AdamJaggard**
Replaces the custom seek-to-live button with the new one from Media Chrome.

Currently using the latest canary because there's a playback issue linking the head of main with the currently deployed media-chrome. @cjpillsbury can you let me know when that fix is deployed and I will update the dependency.

image

Untitled

### CODE REVIEW

File: packages/mux-player/src/template.ts [hunk 1]
```diff
@@ -108,23 +108,20 @@ export const content = (props: MuxTemplateProps) => html`
...
     ${isLiveOrDVR(props)
-      ? html`<button
+      ? html`<media-live-button
           slot="seek-live"
           part="${isLive(props) ? 'top' : 'bottom'} seek-live button"
           disabled="${props.inLiveWindow || !props.hasSrc || props.isDialogOpen}"
           aria-disabled="${(props.inLiveWindow || !props.hasSrc || props.isDialogOpen) && 'true'}"
-          in-live-window="${props.inLiveWindow}"
+          media-time-is-live="${props.inLiveWindow}"
           onclick="${function (this: HTMLButtonElement, evt: Event) {
```
Comment (gkatsev) [inline packages/mux-player/src/template.ts:117]: I don't think we need this onclick anymore, as it does it natively now: https://github.com/muxinc/media-chrome/blob/8c1e730b4baaa316f285d2be719451a1dea55c00/src/js/media-live-button.js#L59-L72

File: packages/mux-player/src/template.ts [hunk 2]
```diff
@@ -108,23 +108,20 @@ export const content = (props: MuxTemplateProps) => html`
         : html``}
     </mux-video>
     ${isLiveOrDVR(props)
-      ? html`<button
```
Comment (gkatsev) [inline packages/mux-player/src/template.ts:111]: I think this is fine for a first pass, but I think ultimately, we'd want it to be in template itself replacing the slot https://github.com/muxinc/elements/blob/main/packages/mux-player/src/media-theme-mux.html

Comment (AdamJaggard) [reply packages/mux-player/src/template.ts:111]: Just had a chat about this with @cjpillsbury, I'm going to move it to the template before merging 🙂

File: packages/mux-player/package.json [hunk 3]
```diff
@@ -52,7 +52,7 @@
   "dependencies": {
     "@mux/mux-video": "0.12.0",
     "@mux/playback-core": "0.15.0",
-    "media-chrome": "0.16.2"
+    "media-chrome": "0.17.1-canary.10-464ac4a"
```
Comment (gkatsev) [inline packages/mux-player/package.json:55]: Marking this as a blocker, so we don't forget.

**Review by gkatsev — CHANGES_REQUESTED** [review]

**Comment by luwes** [comment]
SCR-20230120-gl7

vertical alignment is slightly off. 
the LIVE text is 14px font-size, the title 16px.

I tried to quickly fix it via web inspector but was not as easy as it seems. 
if you change the LIVE text to 16px for a moment you can see the correct alignment but then that label is too big.

I think the solution is to move the LIVE text 1px down but preferably not with position relative top 1px or something hacky like that. maybe line-height or one of the flex properties, vertical-align, ....

File: packages/mux-player/src/media-theme-mux.html [hunk 4]
```diff
@@ -501,7 +501,7 @@
       </template>
       <media-control-bar>
         {{>PlayButton}}
-        <slot name="seek-live"></slot>
+        <media-live-button disabled="{{disabled}}" aria-disabled="{{disabled}}"></media-live-button>
```
Comment (luwes) [inline packages/mux-player/src/media-theme-mux.html:504]: (blocking): use a template partial for this button with a `part` attribute

**Review by luwes — CHANGES_REQUESTED** [review]
overall it looks great.
I'd prefer if we get the vertical 1px and required `part` attribute fixed before merging

File: packages/mux-player/src/media-theme-mux.html [hunk 5]
```diff
@@ -451,6 +452,15 @@
       </media-cast-button>
     </template>
 
+    <template partial="LiveButton">
+      <!-- NOTE: Keeping "seek-live" part for backwards compatibility for now -->
+      <media-live-button
+        part="{{section ?? 'top'}} live seek-live button"
```
Comment (luwes) [inline packages/mux-player/src/media-theme-mux.html:458]: 👍

File: packages/mux-player/src/media-theme-mux.html [hunk 6]
```diff
@@ -261,7 +261,8 @@
       line-height: 24px;
     }
 
-    .title-display {
+    :is(.title-display, media-live-button) {
+      color: inherit;
       font-size: 16px;
```
Comment (luwes) [inline packages/mux-player/src/media-theme-mux.html:266]: overall great but

(blocking): `media-live-button`'s  should be 14px and vertically aligned against the red dot and the title

**Review by luwes — CHANGES_REQUESTED** [review]

**Pull request merged**